### ISSUE TITLE
File corrupted

### ISSUE THREAD

**Issue by hulucc** [mbdavid/LiteDB#1018]
LiteDB v4, some BSON corrupted in <Visit> collection.
How can i figure out what happened?
[service.c.db.zip](https://github.com/mbdavid/LiteDB/files/2144946/service.c.db.zip)

**Comment by mbdavid** [comment]
Hi @hulucc, here your database recovery database

[service.c-recovery.db.zip](https://github.com/mbdavid/LiteDB/files/2153185/service.c-recovery.db.zip)

You can recovery your datafile with current master branch version. Has an static method in LiteEngine to recovery data (and a sample in LiteDB.Demo project).

But, how this happened? How is your environment?

**Comment by lidanger** [comment]
Maybe, it's a good way that creating a new data file, recreate the index and copy all collection items to the new file.It's much faster.

**Comment by hulucc** [comment]
Hi, @mbdavid
Thank you for the information.
It happens on one of my client's computer, only once.
I don't think it can be reproduced...

**Comment by lbnascimento** [comment]
Hi! With the objective of organizing our issues, we are closing old unsolved issues. Please check the latest version of LiteDB and open a new issue if your problem/question/suggestion still applies. Thanks!
